ethash/src/compute.rs

```diff
@@ -85,6 +86,7 @@ pub type H256 = [u8; 32];
 pub struct Light {
 	block_number: u64,
 	cache: Vec<Node>,
+	eth_compute: Mutex<EthCompute>,
```

Comment:
@gavofyork @arkpar Actually that doesn't work. There is already a Mutex in the structure between `Ethash` and `Light`, but it doesn't help.

```Rust
pub struct EthashManager {
    cache: Mutex<LightCache>,
}
```

```
peter$ cargo build
   Compiling ethcore v1.1.0 (file:///Users/peter/dev/rust/ethcore/parity)
ethcore/src/ethereum/ethash.rs:62:6: 62:12 error: the trait `core::marker::Sync` is not implemented for the type `core::cell::UnsafeCell<u64>` [E0277]
ethcore/src/ethereum/ethash.rs:62 impl Engine for Ethash {
                                       ^~~~~~
ethcore/src/ethereum/ethash.rs:62:6: 62:12 help: run `rustc --explain E0277` to see a detailed explanation
ethcore/src/ethereum/ethash.rs:62:6: 62:12 note: `core::cell::UnsafeCell<u64>` cannot be shared between threads safely
ethcore/src/ethereum/ethash.rs:62:6: 62:12 note: required because it appears within the type `core::cell::Cell<u64>`
ethcore/src/ethereum/ethash.rs:62:6: 62:12 note: required because it appears within the type `ethereum::ethash::ethash::compute::SeedHashCompute`
ethcore/src/ethereum/ethash.rs:62:6: 62:12 note: required because it appears within the type `ethereum::ethash::ethash::compute::Light`
ethcore/src/ethereum/ethash.rs:62:6: 62:12 note: required because it appears within the type `core::option::Option<alloc::arc::Arc<ethereum::ethash::ethash::compute::Light>>`
ethcore/src/ethereum/ethash.rs:62:6: 62:12 note: required because it appears within the type `ethereum::ethash::ethash::LightCache`
ethcore/src/ethereum/ethash.rs:62:6: 62:12 note: required because it appears within the type `ethereum::ethash::ethash::EthashManager`
ethcore/src/ethereum/ethash.rs:62:6: 62:12 note: required because it appears within the type `ethereum::ethash::Ethash`
ethcore/src/ethereum/ethash.rs:62:6: 62:12 note: required by `engine::Engine`
error: aborting due to previous error
Could not compile `ethcore`.
```

I do not know how to proceed from here.

Here is the current state of the code, which produces the error above:
https://github.com/peterjoel/parity/tree/ethcompute_no_mutex